trlx/pipeline/offline_pipeline.py

```diff
@@ -52,13 +54,28 @@ class PromptPipeline(BasePipeline):
     Tokenizes prompts, unless they are already tokenized, and truncates them to `max_prompt_length` from the right
     """
 
-    def __init__(self, prompts: List[str], max_prompt_length: int, tokenizer=None):
+    def __init__(self, prompts: List[str], max_prompt_length: int, tokenizer: PreTrainedTokenizer):
         super().__init__()
-        model_inputs = tokenizer(prompts, truncation=True, padding=False, max_length=max_prompt_length)
-        prompts = model_inputs["input_ids"]
+
+        # manually prepend bos token if not already present to match `tokenize_dialog` for ILQL and
+        # default tokenizer behavior for PPO
+        if tokenizer.bos_token is not None:
+            prompts = [
+                tokenizer.bos_token + prompt if not prompt.startswith(tokenizer.bos_token) else prompt
```

Comment:
Do we have to force adding BOS here? In ILQL it is only added in case there is only a single string passed, to make sure loss starts from 0-index of output (action_ixs[0] = 0) by prepending a "prompt" as BOS, and not when real `[prompt, output]` is passed (unless truncation comes into play). Don't want to change behaviour here as I'm finilizing HH PR with already shaby results as they are

Reply:
I see, that makes sense that the `bos` is only added to the the prompt for a completion if no prompt is given for that completion.